examples/qmainwindow/main.cpp

```diff
@@ -0,0 +1,173 @@
+    // because some APIs are not supprted on old systems such as Windows 7
+    // and Windows 8. And once we have set the DPI awareness level in the
+    // manifest file, any attemptation to try to change it through API will
+    // fail. In other words, Qt won't be able to enable or disable high DPI
+    // scaling or change the DPI awareness level once we have set it in the
+    // manifest file. So the following two lines are uesless actually (However,
+    // they are still useful on other platforms).
+    QGuiApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+    QGuiApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
+    // Don't round the scale factor.
+    // This will break QWidget applications because they can't render correctly.
+    // Qt Quick applications won't have this issue.
+    QGuiApplication::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
+#endif
+#endif
+
+    QApplication application(argc, argv);
+
+    QtAcrylicMainWindow *mainWindow = new QtAcrylicMainWindow(NULL, Qt::WindowFlags());
+    mainWindow->setAcrylicEnabled(true);
+
+    Ui::MainWindow appMainWindow;
+    appMainWindow.setupUi(mainWindow);
+
+    QWidget *widget = new QWidget;
+    Ui::TitleBar titleBarWidget;
+    titleBarWidget.setupUi(widget);
+
+    QMenuBar *menuBar = mainWindow->menuBar();
+    titleBarWidget.horizontalLayout->insertWidget(1, menuBar);
+
+    mainWindow->setMenuWidget(widget);
+
+    QObject::connect(mainWindow,
+                     &QMainWindow::windowIconChanged,
+                     titleBarWidget.iconButton,
+                     &QPushButton::setIcon);
+    QObject::connect(mainWindow,
+                     &QMainWindow::windowTitleChanged,
+                     titleBarWidget.titleLabel,
+                     &QLabel::setText);
+    QObject::connect(titleBarWidget.closeButton,
+                     &QPushButton::clicked,
+                     mainWindow,
+                     &QMainWindow::close);
+    QObject::connect(titleBarWidget.minimizeButton,
+                     &QPushButton::clicked,
+                     mainWindow,
+                     &QMainWindow::showMinimized);
+    QObject::connect(titleBarWidget.maximizeButton,
+                     &QPushButton::clicked,
+                     [mainWindow, titleBarWidget]() {
+                         if (mainWindow->isMaximized()) {
+                             mainWindow->showNormal();
+                         } else {
+                             mainWindow->showMaximized();
+                         }
+                     });
+#ifdef WIN32
+    QObject::connect(titleBarWidget.iconButton,
+                     &QPushButton::clicked,
+                     mainWindow,
+                     [mainWindow] () {
+                        HWND hWnd = (HWND)QGuiApplication::platformNativeInterface()->nativeResourceForWindow(QByteArrayLiteral("handle"), mainWindow->windowHandle());
```

Comment:
Why not just use `mainWindow->winId()` or `mainWindow->windowHandle()->winId()`?